### PR TITLE
feat(resource): CloudFront Web ACL Association

### PR DESCRIPTION
**Pull request by greengangsta**
*Issue #, if available:*

*Description of changes:*

Adding a new resource that can be used to associated a WebACL to a CloudFront distribution, when the CloudFront distribution was created outside of the CloudFormation template. Currently, CloudFormation ‘AWS::WAFv2::WebACLAssociation’ resource has a limitation that a new WebACL provisioned using CloudFormation cannot be associated to an existing CloudFront distribution (created outside of the CloudFormation template). For associating a web ACL to resources in CloudFormation, we can create a resource ‘AWS::WAFv2::WebACLAssociation’. However, this resource type only supports below ResourceArn values for Application Load Balancer, API Gateway and AppSync GraphQL API. Please refer to below documentation link : https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-wafv2-webaclassociation.html

To address this, In the handler methods we make API calls to the CloudFront service and update the distribution configuration. The idea is to first fetch the current configuration of the CloudFront distribution by making the get_distribution_config API call , extract the etag value from the response (this value is required to perform the update Distribution API call), make required changes to the current configurations apply the new configuration to the CloudFront distribution by supplying the new configuration object, and etag value in the update_distribution API call.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ericzbeard — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
 .DS_Store
-.tmpenv
-
+cfassociation.yaml
```
Comment (kddejong) [inline .gitignore:2]: Seems like an odd root level gitignore

File: resources/CloudFront_WebACLAssociation/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+cloudformation-cli-python-lib>=2.1.9
```
Comment (kddejong) [inline resources/CloudFront_WebACLAssociation/requirements.txt:1]: @ericzbeard do we need a requirements-dev.txt file too

**Review by kddejong — CHANGES_REQUESTED** [review]
Few questions.

File: .gitignore (hunk 1 above)
Comment (greengangsta) [reply .gitignore:2]: removed the gitignore

**Comment by greengangsta** [comment]
> Few questions.

added code for cicd pipeline and remoted the gitignore

**Review by kddejong — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1,3 +0,0 @@
-.DS_Store
-.tmpenv
```
Comment (kddejong) [inline .gitignore:2]: Little confusion here.  I was worried about the file that was added to the file not the file itself.

Can we bring this file back with

```
.DS_Store
.tmpenv
```

You may also have to delete the `.DS_Store` file above

**Review by kddejong — CHANGES_REQUESTED** [review]

**Review by kddejong — APPROVED** [review]

**Review by kddejong — APPROVED** [review]

**Pull request merged**